source/app/blueprints/case/case_tasks_routes.py

```diff
@@ -256,10 +256,10 @@ def case_edit_task(cur_id, caseid):
 @case_tasks_blueprint.route('/case/tasks/delete/<int:cur_id>', methods=['POST'])
 @ac_api_case_requires(CaseAccessLevel.full_access)
 def case_delete_task(cur_id, caseid):
-    call_modules_hook('on_preload_task_delete', data=cur_id, caseid=caseid)
     task = get_task_with_assignees(task_id=cur_id, case_id=caseid)
     if not task:
         return response_error("Invalid task ID for this case")
+    call_modules_hook('on_preload_task_delete', data=task, caseid=caseid)
```

Comment:
Maybe then we should update all preload deletions to work with a dump of the object? Normally preload hooks are only working with raw data, nothing from the DB, to avoid any conflicts. 
I was acutally thinking of removing all preload in future versions. Do you have any use cases where working with preload instead of postload?

Reply:
In our case, there is no difference between preload and postload. We need to track the deletion of cases and tasks. Since postload works after the deletion of object and it is impossible to process a non-existent object with a webhook module, so I tried to restore the correct operation of preload, of course we need information about the object in the output to work with it.